src/wurf/dependency.py

```diff
@@ -79,34 +91,52 @@ def __init__(self, **kwargs):
         #    in external formats such as files
         # 2. Use the Python hash() function to generate a integer hash for 
         #    run-time usage (i.e. within the same interpreter session).
-        self.hash = None
+        #super(Dependency, self).__setattr__('hash', None)
+        self.read_only['hash'] = None
+        
+        # Resolver attributes (modifiable)
+        object.__setattr__(self, 'read_write', dict())
+        
         
     def __getattr__(self, option):
         """ Return the value corresponding to the option.
 
         :param option: The name of the option to return as a string.
         :return: The option value.
         """
+        if option in self.read_only:
+            return self.read_only[option]
+
+        elif option in self.read_write:
+            return self.read_write[option]
+        
+        else:
+            return None
+
+
+    def __setattr__(self, name, value): 
```

Comment:
Here you used `name` instead of `option`. Again, `property` would feel more accurate.